ruma-api-macros/src/api/request/outgoing.rs

```diff
@@ -131,23 +131,31 @@ impl Request {
             .collect();
 
         for auth in &metadata.authentication {
-            if auth.value == "AccessToken" {
-                let attrs = &auth.attrs;
-                header_kvs.extend(quote! {
+            let attrs = &auth.attrs;
+            let auth_value = &auth.value.to_string();
+            header_kvs.extend(quote! {{
+                let access_token = match access_token {
+                    #ruma_api::SendAccessToken::IfRequired(access_token) => {
+                        if #auth_value == "AccessToken" {
```

Comment:
This is extra code for the compiler to constant-fold for every single invocation of the macro. This check should be outside of `quote!`.

Reply:
So like moving just `#auth_value == "AccessToken"` outside of `quote!` and doing `if boolean_result {` here? Because moving out the whole match doesn't work since it needs the `access_token`

Reply:
The former makes sense anyway, so done that